numba/core/typing/templates.py

```diff
@@ -720,7 +724,56 @@ def _build_impl(self, cache_key, args, kws):
             On failure, returns `(None, None)`.
 
         """
+        from numba.core import decorators
         from numba import jit
+        jitter_str = self.metadata.get('hardware', None)
```

Comment:
We should move this entire `jitter` selection logic out into its own method

Reply:
Agree, done.